src/internet_identity/tests/integration/v2_api/authn_method_metadata.rs

```diff
@@ -0,0 +1,201 @@
+
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let authn_method = test_authn_method();
+    let identity_number = create_identity_with_authn_method(&env, canister_id, &authn_method);
+
+    let identity_info =
+        api_v2::identity_info(&env, canister_id, authn_method.principal(), identity_number)?
+            .expect("identity info failed");
+    let actual_metadata = &identity_info
+        .authn_methods
+        .first()
+        .expect("expect authn_methods not to be empty")
+        .metadata;
+    assert!(actual_metadata.is_empty());
+
+    let metadata = HashMap::from([(
+        METADATA_KEY.to_string(),
+        MetadataEntryV2::String("some value".to_string()),
+    )]);
+
+    api_v2::authn_method_metadata_replace(
+        &env,
+        canister_id,
+        authn_method.principal(),
+        identity_number,
+        &authn_method.public_key(),
+        &metadata,
+    )?
+    .expect("identity metadata replace failed");
+
+    let identity_info =
+        api_v2::identity_info(&env, canister_id, authn_method.principal(), identity_number)?
+            .expect("identity info failed");
+    let actual_metadata = &identity_info
+        .authn_methods
+        .first()
+        .expect("expect authn_methods not to be empty")
+        .metadata;
+    assert_eq!(actual_metadata, &metadata);
+    Ok(())
+}
+
+#[test]
+fn should_require_authentication_to_replace_identity_metadata() {
+    const METADATA_KEY: &str = "some-key";
+
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let authn_method = test_authn_method();
+    let identity_number = create_identity_with_authn_method(&env, canister_id, &authn_method);
+
+    let metadata = HashMap::from_iter(vec![(
+        METADATA_KEY.to_string(),
+        MetadataEntryV2::String("some value".to_string()),
+    )]);
+
+    let result = api_v2::authn_method_metadata_replace(
+        &env,
+        canister_id,
+        Principal::anonymous(),
+        identity_number,
+        &authn_method.public_key(),
+        &metadata,
+    );
+    expect_user_error_with_message(
+        result,
+        CanisterCalledTrap,
+        Regex::new("[a-z\\d-]+ could not be authenticated.").unwrap(),
+    );
+}
+
+#[test]
+fn should_not_write_too_large_identity_metadata_map() -> Result<(), CallError> {
+    const METADATA_KEY: &str = "some-key";
+
+    let env = env();
+    let canister_id = install_ii_canister(&env, II_WASM.clone());
+    let authn_method = test_authn_method();
+    let identity_number = create_identity_with_authn_method(&env, canister_id, &authn_method);
+
+    let identity_info =
+        api_v2::identity_info(&env, canister_id, authn_method.principal(), identity_number)?
+            .expect("identity info failed");
+    assert!(identity_info.metadata.is_empty());
+
+    let metadata = HashMap::from([(
+        METADATA_KEY.to_string(),
+        MetadataEntryV2::String("a".repeat(3000)),
```

Comment:
maybe use a value closer to `VARIABLE_FIELDS_LIMIT` (and mention the constant name in a comment)? or export the constant and use it also in the test?